internal/internal.go

```diff
@@ -148,34 +162,49 @@ func (p *envoyExtAuthzGrpcServer) Check(ctx ctx.Context, req *ext_authz.CheckReq
 
 	logrus.WithFields(logrus.Fields{
 		"query":               p.cfg.Query,
-		"decision":            result,
+		"decision":            result.decision,
 		"err":                 err,
-		"txn":                 txnID,
-		"metrics":             metrics.All(),
+		"txn":                 result.txnID,
+		"metrics":             result.metrics.All(),
 		"total_decision_time": time.Since(start),
 	}).Info("Returning policy decision.")
 
+	err = p.log(ctx, input, result, err)
```

Comment:
If the decision logger returns an error, the logrus message above is going to be confusing. I think we ought to make them consistent.

Reply:
Done.